sound/soc/soc-pcm.c

```diff
@@ -2963,9 +2963,9 @@ int soc_new_pcm(struct snd_soc_pcm_runtime *rtd, int num)
 
 	pcm->no_device_suspend = true;
 out:
-	dev_info(rtd->card->dev, "%s <-> %s mapping ok\n",
-		 (rtd->num_codecs > 1) ? "multicodec" : asoc_rtd_to_codec(rtd, 0)->name,
-		 (rtd->num_cpus > 1)   ? "multicpu"   : asoc_rtd_to_cpu(rtd, 0)->name);
+	dev_dbg(rtd->card->dev, "%s <-> %s mapping ok\n",
+		(rtd->num_codecs > 1) ? "multicodec" : asoc_rtd_to_codec(rtd, 0)->name,
+		(rtd->num_cpus > 1)   ? "multicpu"   : asoc_rtd_to_cpu(rtd, 0)->name);
```

Comment:
This one is helpful for me. I can see which rtd is mapping ok with dynamic debug disabled. So I will know which rtd may have problem before dynamic debug is enabled.

Reply:
@bardliao I have personally never used this information, I don't even know what it means. Can you clarify what sort of error classes can be identified with this trace?

Reply:
@bardliao If you look at the code in soc_pcm_new, what you really want is a better message when the following code fails

````c
	ret = snd_soc_pcm_component_new(rtd);
	if (ret < 0) {
		dev_err(rtd->dev, "ASoC: pcm constructor failed: %d\n", ret); <<< HERE
		return ret;
	}

	pcm->no_device_suspend = true;
out:
	dev_info(rtd->card->dev, "%s <-> %s mapping ok\n",
		 (rtd->num_codecs > 1) ? "multicodec" : asoc_rtd_to_codec(rtd, 0)->name,
		 (rtd->num_cpus > 1)   ? "multicpu"   : asoc_rtd_to_cpu(rtd, 0)->name);
	return ret;
````

Reply:
> @bardliao If you look at the code in soc_pcm_new, what you really want is a better message when the following code fails. Having a trace when everything works fine is not really useful.
> 
> ```c
> 	ret = snd_soc_pcm_component_new(rtd);
> 	if (ret < 0) {
> 		dev_err(rtd->dev, "ASoC: pcm constructor failed: %d\n", ret); <<< HERE
> 		return ret;
> 	}
> 
> 	pcm->no_device_suspend = true;
> out:
> 	dev_info(rtd->card->dev, "%s <-> %s mapping ok\n",
> 		 (rtd->num_codecs > 1) ? "multicodec" : asoc_rtd_to_codec(rtd, 0)->name,
> 		 (rtd->num_cpus > 1)   ? "multicpu"   : asoc_rtd_to_cpu(rtd, 0)->name);
> 	return ret;
> ```

Thanks @plbossart You are right.